pkg/monitoring/rules/rules.go

```diff
@@ -0,0 +1,54 @@
+package rules
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/monitoring/rules/alerts"
+	"kubevirt.io/containerized-data-importer/pkg/monitoring/rules/recordingrules"
+)
+
+const (
+	ruleName = "prometheus-cdi-rules"
+)
+
+// SetupRules initializes recording and alert rules in a namespace.
+func SetupRules(namespace string) error {
+	if err := recordingrules.Register(namespace); err != nil {
+		return err
+	}
+
+	if err := alerts.Register(namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// BuildPrometheusRule creates a PrometheusRule in a namespace.
+func BuildPrometheusRule(namespace string) (*promv1.PrometheusRule, error) {
+	rules, err := operatorrules.BuildPrometheusRule(
+		ruleName,
+		namespace,
+		map[string]string{
+			common.CDIComponentLabel:  "",
+			common.PrometheusLabelKey: common.PrometheusLabelValue,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return rules, nil
```

Comment:
you can just return `operatorrules.BuildPrometheusRule...`

Reply:
Not sure since they expect *promv1.PrometheusRule:
"Cannot use 'operatorrules.BuildPrometheusRule' (type func(name string, namespace string, labels map[string]string) (*promv1.PrometheusRule, error)) as the type *promv1.PrometheusRule"

Reply:
could also be improved there

`operatorrules.BuildPrometheusRule` already returns `rules, error`
`return operatorrules.BuildPrometheusRule, err` should be return `operatorrules.BuildPrometheusRule`

Reply:
thanks a lot! fixed

Reply:
I think you forgot to update here